QA page details

emma-sg opened the pull request:
- Closes #1651.
- Closes https://github.com/webrecorder/browsertrix-cloud/issues/1508

### Changes
- [x] improves overflow issues at smaller screen sizes
- [x] adds icons to buttons
- [x] updates text & layout to match mocks
- [x] changes primary button & button options depending on if there's a qa run available
- [x] adds a loading state for qa run status & buttons
- [x] updates `<btrix-crawl-status>` with a `type` param allowing for crawls, uploads, and QA runs
- [x] Updates `<btrix-alert>` to match `<sl-tag>` styling
- [x] Improves overflow issues at smaller viewport sizes by making tab lists overflow when necessary

### Features
- [x] Ability to start/stop/cancel QA runs https://github.com/webrecorder/browsertrix-cloud/pull/1666 @SuaYoo 
- [x] Ability to see progress of current QA run @emma-sg 
- [x] Ability to delete QA runs @emma-sg 
- [x] Ability to download QA run files https://github.com/webrecorder/browsertrix-cloud/pull/1666 @SuaYoo 
- [x] Only able to start review if a QA Run is finished (for now, initial pass). @SuaYoo 
- [x] Only most recent running or successful QA run is displayed in header

I'd like to refactor a bunch of this code in a follow-up PR once this is merged.

emma-sg committed:
first pass at qa details
- improves overflow issues at smaller screen sizes
- adds icons to buttons
- updates text & layout to match mocks
- changes primary button & button options depending on if there's a qa run available
- adds a loading state for qa run status & buttons
- updates <btrix-crawl-status> with a `type` param allowing for crawls, uploads, and QA runs

emma-sg committed:
first pass at qa details
- improves overflow issues at smaller screen sizes
- adds icons to buttons
- updates text & layout to match mocks
- changes primary button & button options depending on if there's a qa run available
- adds a loading state for qa run status & buttons
- updates <btrix-crawl-status> with a `type` param allowing for crawls, uploads, and QA runs

SuaYoo commented on frontend/src/features/archived-items/crawl-status.ts:
```diff
@@ -155,11 +158,15 @@ export class CrawlStatus extends LitElement {
+        label = {
+          upload: msg("Uploaded"),
+          crawl: msg("Complete"),
+          qa: msg("Ready for Review"),
```
I believe you were out when we discussed this, sorry that notes are missing, but we're decoupling the QA analysis and QA review statuses, so this can just be "complete".
```suggestion
          qa: msg("Complete"),
```

emma-sg replied:
Ah cool, thanks! Were there any other changes from the mockups I should be aware of? cc @Shrinks99

emma-sg committed:
Update frontend/src/features/archived-items/crawl-status.ts

Change "Ready for Review" to "Complete"

Co-authored-by: sua yoo <sua@webrecorder.org>

emma-sg committed:
Update frontend/src/features/archived-items/crawl-status.ts

Change "Ready for Review" to "Complete"

Co-authored-by: sua yoo <sua@webrecorder.org>

SuaYoo committed:
Update QA tab page list and standardize review statuses (#1662)

- Paginates QA tab page list
- Sort QA tab page list by title, URL, comments, and approval; displays
all sortable fields
- Standardizes page "reviews" -> "approvals"
- Standardizes crawl review rating labels
- Displays only latest comment in review page list

---------

Co-authored-by: emma <hi@emma.cafe>

SuaYoo committed:
Update QA tab page list and standardize review statuses (#1662)

- Paginates QA tab page list
- Sort QA tab page list by title, URL, comments, and approval; displays
all sortable fields
- Standardizes page "reviews" -> "approvals"
- Standardizes crawl review rating labels
- Displays only latest comment in review page list

---------

Co-authored-by: emma <hi@emma.cafe>

SuaYoo committed:
refactor qa tab into component

SuaYoo committed:
refactor qa tab into component

SuaYoo committed:
disable buttons when analysis is running

SuaYoo committed:
disable buttons when analysis is running

emma-sg committed:
Merge branch 'main' into frontend-qa-overview-page

emma-sg committed:
fix lint issues

emma-sg committed:
fix lint issues

emma-sg committed:
more overflow and wrapping fixes

emma-sg committed:
more overflow and wrapping fixes

SuaYoo committed:
fix rating

emma-sg committed:
Merge branch 'frontend-qa-overview-page' of https://github.com/webrecorder/browsertrix-cloud into frontend-qa-overview-page

emma-sg committed:
fix some more small spacing & overflow issues

emma-sg committed:
restrict review status to set number literals

emma-sg committed:
add titles to links so that full titles/urls can be viewed

SuaYoo committed:
Increase QA review rating scale (#1668)

Related https://github.com/webrecorder/browsertrix-cloud/issues/1663

Increases rating scale from 3 to 5.

---------
Co-authored-by: emma <hi@emma.cafe>

emma-sg committed:
Merge branch 'main' into frontend-qa-overview-page

SuaYoo committed:
Stop, cancel, and download QA run (#1666)

Partially addresses
https://github.com/webrecorder/browsertrix-cloud/issues/1651

### Changes

- Disables review actions if analysis is running
- Enables downloading run
- Enables stopping and canceling runs
- Minor refactor to create and implement `btrix-menu-item-link` for
download menu option
- Hides canceled QA runs from dropdown

---------
Co-authored-by: emma <hi@emma.cafe>

emma-sg committed:
add progress bar

emma-sg committed:
add tooltip and min widths to loaders

emma-sg committed:
move section to separate desc list item

emma-sg committed:
re-fetch periodically when starting or running qa

emma-sg committed:
improve tooltip text when loading pages

emma-sg committed:
show most recent successful qa run info

emma-sg committed:
make cancel dialog buttons push more towards not cancelling

emma-sg committed:
only disable qa button when there's no available qa runs

emma-sg committed:
wip delete qa

emma-sg committed:
fix qa availability for header buttons

emma-sg committed:
ensure browser navigation corresponds to active tab

ikreymer committed:
Merge branch 'main' into frontend-qa-overview-page

ikreymer committed:
Merge branch 'frontend-qa-overview-page' of github.com:webrecorder/browsertrix-cloud into frontend-qa-overview-page

emma-sg committed:
complete delete confirmation dialog

emma-sg committed:
use str tag for localized string

emma-sg committed:
identify separate pages translations

SuaYoo commented on frontend/src/components/ui/desc-list.ts:
```diff
@@ -30,7 +30,7 @@ export class DescListItem extends LitElement {
       font-size: var(--sl-font-size-x-small);
       line-height: 1rem;
-      margin: var(--sl-spacing-3x-small) 0;
+      margin: 0 0 0.25rem 0;
```
Does this need to be custom spacing?

emma-sg replied:
Oh, nope, was just doing tweaking in devtools and copied the value — happy to change it back

emma-sg replied:
Addressed in 0b77bcbcbba2751bbaf6da3f66828eb9ef83183b.

SuaYoo commented on frontend/src/components/ui/tab-list.ts:
```diff
@@ -247,13 +247,13 @@ export class TabList extends LitElement {
           </div>
 
-          <ul class="tablist" role="tablist">
+          <ul class="tablist -m-3 overflow-x-auto p-3" role="tablist">
```
This seems to be causing horizontal scroll, testing with firefox on, present on all screen sizes with two-column

<img width="296" alt="Screenshot 2024-04-16 at 1 54 51 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/de076bd7-e3da-4f36-970e-d30f592b2809">

emma-sg replied:
This _is_ an issue, but it's not in `frontend/src/components/ui/tab-list.ts` — this page doesn't use tab-list at all. Fixed in c0969506546567989af76b79c13357262658224c.

SuaYoo commented on frontend/src/pages/org/archived-item-detail/ui/qa.ts:
```diff
@@ -1,216 +1,718 @@
+        >
+        <div>
+          ${msg(
+            str`This QA run includes data for ${runToBeDeleted.stats.done} ${pluralize(runToBeDeleted.stats.done, { zero: msg("pages", { desc: 'plural form of "page" for zero pages', id: "pages.plural.zero" }), one: msg("page"), two: msg("pages", { desc: 'plural form of "page" for two pages', id: "pages.plural.two" }), few: msg("pages", { desc: 'plural form of "page" for few pages', id: "pages.pl ...
```
Love the additional context!

SuaYoo commented on frontend/src/pages/org/archived-item-detail/ui/qa.ts:
```diff
@@ -1,216 +1,718 @@
+    if (!runToBeDeleted) return;
+
+    return html`
+      <btrix-dialog id="deleteQARunDialog" .label=${msg("Delete QA Run?")} open>
```
```suggestion
      <btrix-dialog id="deleteQARunDialog" .label=${msg("Delete QA Run?")}>
```

Shrinks99 replied:
Another comment for this one, we've been calling these both "QA Runs", "Analysis Runs", and "Crawl Analysis".  I'm partial to referencing it as "Crawl Analysis" as I think it requires the least additional explanation, what are the units of crawl analysis?  When the "Crawl Analysis" process completes, what does it output?

Whatever that is, we should refer to it here!  Some options:
- Crawl Analysis Runs: "delete crawl analysis run"
- Crawl Analysis: "delete crawl analysis"
- Crawl Analysis Instance: "delete crawl analysis instance"

SuaYoo replied:
Could be "delete analysis run" since the list is labeled "Analysis Runs"? I just renamed the download button to match.

emma-sg replied:
Addressed in 2bc2e0e

SuaYoo commented on frontend/src/pages/org/archived-item-detail/ui/qa.ts:
```diff
@@ -1,216 +1,718 @@
+  private readonly renderDeleteConfirmDialog = () => {
+    const runToBeDeleted = this.qaRuns?.find((run) => run.id === this.deleting);
+
+    if (!runToBeDeleted) return;
```
This causes the dialog backdrop to immediately render into view instead of transitioning smoothly the first time it's opened, could either make the content of `btrix-dialog` render conditionally, or switch to `runToBeDeleted?[prop]`

emma-sg replied:
Ahh, ty!

emma-sg replied:
Addressed in 2bc2e0e

SuaYoo commented on frontend/src/pages/org/archived-item-detail/ui/qa.ts:
```diff
@@ -1,216 +1,718 @@
+        <div slot="footer" class="flex justify-between">
+          <sl-button
+            size="small"
+            variant="primary"
```
Delete confirmation button variants are a bit all over the place in the app, this might be a good place to start standardizing-- @Shrinks99 what should the button variants be for "Cancel" and "Delete" in the confirmation dialog?

Shrinks99 replied:
Cancel and Delete are both destructive actions.  We've been using red for those, would use the danger variant for both,  filled if it's the primary action, unfilled line option if it's not.  Would like to highlight that the action they intend to take is dangerous, but still the primary one that they intended.

Cancel in this dialog should be the standard button type as it does not correspond with the action they intended to take opening the dialog.

emma-sg replied:
Sure, makes sense!

emma-sg replied:
Addressed in 2bc2e0e

SuaYoo commented on frontend/src/pages/org/archived-item-detail/archived-item-detail.ts:
```diff
@@ -198,60 +242,28 @@ export class ArchivedItemDetail extends TailwindElement {
     switch (this.activeTab) {
       case "qa":
         sectionContent = this.renderPanel(
-          html`${this.renderTitle(msg("Quality Assurance (QA)"))}
```
I added "(QA)" since we don't otherwise specify in the app what QA abbreviates. If the move is to remove (QA), can we add an info popup somewhere that provides the expansion of the abbreviation? cc @Shrinks99

Shrinks99 replied:
Can we just rename the label to `Quality Assurance`?  Seems to be enough room for it!

If not, I think having the label on the page "Quality Assurance" provides the expansion we are looking for?

SuaYoo replied:
+1 for naming tab "Quality Assurance"

SuaYoo reviewed (changes requested):
Looking good! Minor feedback.

Shrinks99 commented on frontend/src/features/qa/page-list/page-list.ts:
```diff
@@ -28,9 +32,13 @@ const sortableFields = {
   },
   approved: {
-    label: msg("Review Status"),
+    label: msg("Approval"),
```
Not sure if this is the most accurate title for the column?  "Page status?"  I liked "Review Status"...

SuaYoo replied:
We discussed "Review" referring to reviewing the whole crawl, since the process of approving/rejecting is called an approval process.

Shrinks99 replied:
Looks like I've been bitten by my own problem of not keeping the mock-ups up to date this round!  Keep as is if I've just forgotten. 👍

SuaYoo committed:
rename download button

SuaYoo commented:
I noticed that the QA runs dropdown and tab doesn't update after a run is complete. Looking now.

SuaYoo committed:
tag current analysis run

SuaYoo committed:
show alert

SuaYoo committed:
fix double fetch

emma-sg committed:
use shoelace spacing vars in desc-list

emma-sg committed:
fix sidebar tab overflow issue

emma-sg committed:
Merge branch 'frontend-qa-overview-page' of https://github.com/webrecorder/browsertrix-cloud into frontend-qa-overview-page

SuaYoo commented:
Added tab and alert message:

<img width="1045" alt="Screenshot 2024-04-17 at 10 57 13 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/0a600705-d9ea-4aa4-bfb5-19edb32d2d9d">
<img width="1033" alt="Screenshot 2024-04-17 at 10 57 19 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/dcbb9a8a-dfec-4509-b33c-383949a08ba2">
<img width="1041" alt="Screenshot 2024-04-17 at 11 02 21 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/02ea3438-4de4-40d9-84a5-0647b5572a96">

SuaYoo committed:
update to most recently run once completed

SuaYoo commented:
> I noticed that the QA runs dropdown and tab doesn't update after a run is complete. Looking now.

addressed in https://github.com/webrecorder/browsertrix-cloud/pull/1656/commits/28709a12bb0be5e603823ec8ccd581778c012998 @emma-sg

emma-sg committed:
analysis deletion dialog improvements
- make analysis run naming consistent
- allow deletion dialog to always render, allowing for animation on initial enter
- make button styling in dialog consistent, using filled danger style for delete and neutral style for cancel

emma-sg committed:
Merge branch 'frontend-qa-overview-page' of https://github.com/webrecorder/browsertrix-cloud into frontend-qa-overview-page

emma-sg commented:
@SuaYoo re: tab and alert messages, should we unify the styling of the tag and alert warnings? I feel like it'd make sense for both to use the same amber colours and borders etc. cc @Shrinks99 
<img width="487" alt="Screenshot 2024-04-17 at 2 43 12 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5727389/54fdacb6-02c1-42a6-a95a-5bd24abc7f28">

SuaYoo commented:
> @SuaYoo re: tab and alert messages, should we unify the styling of the tag and alert warnings?

Yep that should be pretty quick change to `alert.ts` if you want to go ahead and do that!

Shrinks99 committed:
Icon consistency fix

emma-sg committed:
make alert styling consistent with shoelace tags

emma-sg committed:
Merge branch 'frontend-qa-overview-page' of https://github.com/webrecorder/browsertrix-cloud into frontend-qa-overview-page

emma-sg committed:
hide incomplete qa run overview cards

SuaYoo committed:
hide entire section

SuaYoo committed:
fix elapsed time label

Shrinks99 reviewed (approved):
Looks good to me!
